File: src/main/java/com/github/stokito/IdeaSingletonInspection/smells/ClassFinalModifier.java
package com.github.stokito.IdeaSingletonInspection.smells;

import com.intellij.psi.PsiClass;
import com.intellij.psi.PsiModifierList;
import org.jetbrains.annotations.NotNull;

import static com.github.stokito.IdeaSingletonInspection.quickFixes.QuickFixes.SET_CLASS_FINAL;

public class ClassFinalModifier extends Smell {

  @Override
  public void check(@NotNull final PsiClass aClass) {
    final PsiModifierList modifierList = aClass.getModifierList();
    assert modifierList != null;
    if (!modifierList.hasModifierProperty("final")) {
      getHolder().registerProblem(aClass, "Class should be final", SET_CLASS_FINAL);
    }
  }

}

File: src/main/java/com/github/stokito/IdeaSingletonInspection/smells/ClassFinalModifier.kt
package com.github.stokito.IdeaSingletonInspection.smells

import com.github.stokito.IdeaSingletonInspection.quickFixes.QuickFixes
import com.intellij.psi.PsiClass
import com.intellij.psi.PsiModifier.FINAL

class ClassFinalModifier : Smell() {
    override fun check(aClass: PsiClass) {
        val modifierList = aClass.modifierList!!
        if (!modifierList.hasModifierProperty(FINAL)) {
            holder!!.registerProblem(aClass, "Class should be final", QuickFixes.SET_CLASS_FINAL)
        }
    }

}